### ISSUE TITLE
[Bug]: Undo scrolls view away from last change if last change took place independently of caret position

### ISSUE THREAD

**Issue by Slapbox** [ueberdosis/tiptap#4243]
### Which packages did you experience the bug in?

extension-history

### What Tiptap version are you using?

2.1.0-rc.11

### What’s the bug you are facing?

We implement a delete button on certain nodes which does not change the `selection.from` position in the editor, but still when the user uses the `undo` command the view is scrolled to the caret, which can be extremely disruptive

### What browser are you using?

Chrome

### What did you expect to happen?

Pressing undo should not affect the scroll position of the editor in these circumstances. It can scroll the user potentially hundreds of lines away from the last action they took with no way to get back but to manually relocate it.

### Did you update your dependencies?

- [X] Yes, I’ve updated my dependencies to use the latest version of all packages.

### Are you sponsoring us?

- [ ] Yes, I’m a sponsor. 💖

**Comment by Slapbox** [comment]
This is caused by ProseMirror itself and its invocation of `scrollIntoView` during undo operations.

**Comment by bdbch** [comment]
If it's a Prosemirror problem would it make sense to move this issue upstream to the prosemirror repository @Slapbox?

**Comment by Slapbox** [comment]
Yes probably but I lack the ability to prove that it's not TipTap so my issues filed on ProseMirror never go anywhere For example the most recent issue I filed was related to the syntax selection codeblock, related to this issue https://github.com/ueberdosis/tiptap/issues/4255

I felt very confident this was a ProseMirror problem since I traced it to specific ProseMirror code, but Marijnh asked for a repro not dependent on TipTap, and when I tried to create one (more specifically, found an existing one,) I found the issue did not seem possible to replicate in pure ProseMirror: https://codesandbox.io/s/prosemirror-lee-forked-pnq5ny

Without more ProseMirror expertise I find myself unable to adequately repro these issues to get them fixed in ProseMirror. Unfortunately that leaves me with the possibilities of filing them with TipTap or using `patch-package` to fix it for myself, but in that case I've left everyone else with the problem still.
